fs/operations/copy.go

```diff
@@ -97,7 +98,12 @@ func (c *copy) checkPartial() (remoteForCopy string, inplace bool, err error) {
 	}
 	// Avoid making the leaf name longer if it's already lengthy to avoid
 	// trouble with file name length limits.
-	suffix := "." + random.String(8) + c.ci.PartialSuffix
+
+	// generate a stable random suffix: use the significant bytes of the hashed remote
+	hashBytes := sha256.Sum256([]byte(c.remote))
```

Comment:
Making the suffix depend on the remote name will make it identical for all users. This means that if A and B are uploading at the same time then the resultant file won't be either A or B but a corrupted mix of both.

You could use `fs.Fingerprint(ctx, c.src, true)` to create a string which depends on the source in a reasonably robust fashion. I'm not sure this will solve your problem though.

Reply:
Thanks, I'll consider it when we work on the resume part.
I removed all resume-related changes from this PR, including the stable partial filename.